amespahdbpythonsuite/spectrum.py

```diff
@@ -144,10 +144,37 @@ def plot(self, **keywords):
 
     def coadd(self, weights=None, average=False):
         """
-        Co-add spectra.
+        Co-add PAHdb spectra.
+
+        Parameters:
+            weights: dict
+                Dictonary of fit weights to use when co-adding.
+            average: bool
+                If True calculates the average coadded spectrum.
+
         """
+        coadded = dict()
+        coadded['type'] = self.type
+        coadded['version'] = self.version
+        coadded['pahdb'] = self.pahdb
+        coadded['uids'] = list(self.data.keys())
+        coadded['model'] = self.model
+        coadded['units'] = self.units
+        coadded['shift'] = self._shift
+        coadded['grid'] = self.grid
+        coadded['profile'] = self.profile
+        coadded['fwhm'] = self.fwhm
+        coadded['weights'] = weights
+        coadded['averaged'] = average
+        coadded['data'] = np.zeros(len(self.grid))
+
         if weights:
-            coadded = np.zeros(len(self.grid))
             for key in self.data.keys():
-                coadded += self.data[key] * weights[key]
-            return coadded
+                coadded['data'] += self.data[key] * weights[key]
+        else:
+            for key in self.data.keys():
+                coadded['data'] += self.data[key]
+        if average:
+            coadded['data'] /= len(self.data.keys())
+
+        return coadded
```

Comment:
It should return an object of type Coadd.

```
return Coadd(type=self.type,
                        version=self.version,
                        data=coadded,
                        average=average,
                        pahdb=self.pahdb,
                        uids=self.uids,
                        model=self.model,
                        units=self.units,
                        shift=self._shift,
                        grid=self.grid
                        profile=self.profile,
                        fwhm=self.fwhm)
```

Reply:
By the Coadd object you mean calling a Coadd class when it will be implemented?

Reply:
Correct. This class would be the counterpart of the IDL implementation at [coadded](https://github.com/PAHdb/AmesPAHdbIDLSuite/blob/master/src/amespahdbidlsuite_coadded_spectrum__define.pro).